modules/metering-reports.adoc

```diff
@@ -0,0 +1,332 @@
+If `reportingStart` is left unset, the Report will run at the next full reportingPeriod after the time the report is created.
+
+As an example of how to use this field, if you had data already collected dating back to January 1st, 2019 which you wanted to be included in your Report, you could create a report with the following values:
+
+----
+apiVersion: metering.openshift.io/v1
+kind: Report
+metadata:
+  name: pod-cpu-request-hourly
+spec:
+  query: "pod-cpu-request"
+  schedule:
+    period: "hourly"
+  reportingStart: "2019-01-01T00:00:00Z"
+----
+
+
+== reportingEnd
+
+To configure a Report to only run until a specified time, you can set the `spec.reportingEnd` field to an link:https://tools.ietf.org/html/rfc3339#section-5.8[RFC3339 timestamp].
+The value of this field will cause the Report to stop running on its schedule after it has finished generating reporting data for the period covered from its start time until `reportingEnd`.
+Because a schedule will most likely not align with reportingEnd, the last period in the schedule will be shortened to end at the specified reportingEnd time.
+If left unset, then the Report will run forever, or until a `reportingEnd` is set on the Report.
+
+For example, if you wanted to create a report that runs once a week for the month of July:
+
+----
+apiVersion: metering.openshift.io/v1
+kind: Report
+metadata:
+  name: pod-cpu-request-hourly
+spec:
+  query: "pod-cpu-request"
+  schedule:
+    period: "weekly"
+  reportingStart: "2019-07-01T00:00:00Z"
+  reportingEnd: "2019-07-31T00:00:00Z"
+----
+
+== runImmediately
+
+Set `spec.runImmediately` to `true` to run the report immediately with all available data, regardless of the `reportingStart` or `reportingEnd` values, and without checking if there is any data for the report period.
+For reports with a schedule set, it will not wait for each period's reportingEnd to elapse before processing and all reportPeriods between `reportingStart` and `reportingEnd`.
+
+== inputs
+
+The `spec.inputs` field of a Report can be used to override or set values defined in a `ReportQuery`'s `spec.inputs` field.
+
+It is a list of name-value pairs:
+
+----
+spec:
+  inputs:
+  - name: "NamespaceCPUUsageReportName"
+    value: "namespace-cpu-usage-hourly"
+----
+
+The `name` of an input must exist in the ReportQuery's `inputs` list.
+The `value` of the input must be the correct type for the input's `type`.
+
+// TODO(chance): include modules/metering-reportquery-inputs.adoc module
+
+== Roll-up Reports
+
+Report data is stored in the database much like metrics themselves, and can thus be used in aggregated or roll-up reports. A simple use case for a roll-up report is to spread the time required to produce a report over a longer period of time: instead of requiring a monthly report to query and add all data over an entire month, the task can be split into daily reports that each run over a thirtieth of the data.
```

Comment:
themselves and, therefore, can be used
; instead of :

Reply:
To clarify `; instead of :` should I have the `:` after of? Eg:
> A simple use case for a roll-up report is to spread the time required to produce a report over a longer period of time; instead of: requiring a monthly report to query and add all data over an entire month, the task can be split into daily reports that each run over a thirtieth of the data.

Reply:
Sorry for not being clear. I think the text should be:

> A simple use case for a roll-up report is to spread the time required to produce a report over a longer period of time; instead of requiring a monthly report to query and add all data over an entire month, the task can be split into daily reports that each run over a thirtieth of the data.